nightfall-optimist/src/services/database.mjs

```diff
@@ -366,6 +376,44 @@ export async function getMempoolTxsSortedByFee() {
     .toArray();
 }
 
+/**
+ * Find a transaction in the mempool, given some filters
+ */
+async function getMempoolTransaction(query) {
```

Comment:
I am wondering if passing a whole `query` as a parem is not too generic. Maybe we can make more like for-the-purpose abstractions here?

Reply:
Here the query means filter I think. You can pass all the filters needed by the mempool transaction you want to query. I see it's reused in other queries for a mempool transaction.

Reply:
That's right, have used `query` as a way to pass a bunch of filters for mempool-specific queries.

I chose `query` over `filters` to be consistent with the rest of the file in terms of language.

Agree we can do much better... I was just trying to remove some of the repetition without touching too much code.